homeassistant/components/http/__init__.py

```diff
@@ -116,7 +116,21 @@
     is_ban_enabled = conf[CONF_IP_BAN_ENABLED]
     login_threshold = conf[CONF_LOGIN_ATTEMPTS_THRESHOLD]
 
-    if api_password is not None:
+    if api_password == 'welcome':
+        _LOGGER.warning('Using default password, this is not recommended!')
+        hass.components.persistent_notification.async_create(
+            """It is strongly recommended to set a non default password!
+            Edit the password in `secrets.yaml`""",
+            title="Default Password used!")
+    elif api_password is None:
+        _LOGGER.warning('No password set in configuration.yaml http: section!')
+        hass.components.persistent_notification.async_create(
+            """It is strongly recommended to set a password!
+            Edit the password in `secrets.yaml` and include it in
```

Comment:
Nowadays the default configuration that ships with Home Assistant defaults to using `secrets.yaml` for passwords.

However, AFAIK we still have a lot of people from "the old days" that have the password stored in `configuration.yaml`. This comment would confuse them. I think it would be better just to add a link to home-assistant.io that explains it in detail. For example https://home-assistant.io/docs/configuration/secrets/

Reply:
will do